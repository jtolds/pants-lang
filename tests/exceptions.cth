try {
  print "hello"
  try {
    print "there"
    throw "ex1"
    print "notreached"
  } { |e|
    print "inner-handler" e
    throw (+ e "-2")
  }
  print "notreached"
} { |e|
  print "outer-handler" e
}

print (try {
  3
} { |e|
  4
})

print (try {
  3
  throw 4
} { |e|
  e
})

print (try {
  3
  throw "error"
} { |e|
  5
})


throw "leave program"
print "notreached"
