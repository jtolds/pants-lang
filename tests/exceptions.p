try {
  println "hello"
  try {
    println "there"
    throw "ex1"
    println "notreached"
  } { |e|
    println "inner-handler" e
    throw (+ e "-2")
  }
  println "notreached"
} { |e|
  println "outer-handler" e
}

println (try {
  3
} { |e|
  4
})

println (try {
  3
  throw 4
} { |e|
  e
})

println (try {
  3
  throw "error"
} { |e|
  5
})

try {
  3(4)
} { |e|
  println "caught exception! it was:" e
}

try {
  (+ 4 null)
} { |e|
  println "tried adding 4 to null:" e
}

try {
  (+ { "function" } 4.0 )
} { |e|
  println "tried adding closure to float:" e
}

throw "leave program"
println "notreached"
