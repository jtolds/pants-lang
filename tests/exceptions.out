hello
there
inner-handler ex1
outer-handler ex1-2
3
4
5
Exception thrown!
=> leave program

return code: 1
